contracts/BIFI/strategies/CompoundV2/StrategyCompoundV2Solidly.sol

```diff
@@ -0,0 +1,428 @@
+
+        emit ChargedFees(callFeeAmount, beefyFeeAmount, strategistFeeAmount);
+    }
+
+    // swap rewards to {want}
+    function swapRewards() internal {
+        uint256 outputBal = IERC20(output).balanceOf(address(this));
+        ISolidlyRouter(unirouter).swapExactTokensForTokens(outputBal, 0, outputToWantRoute, address(this), block.timestamp);
+    }
+
+    /**
+     * @dev Withdraws funds and sends them back to the vault. It deleverages from market first,
+     * and then deposits again after the withdraw to make sure it mantains the desired ratio.
+     * @param _amount How much {want} to withdraw.
+     */
+    function withdraw(uint256 _amount) external {
+        require(msg.sender == vault, "!vault");
+
+        uint256 wantBal = availableWant();
+
+        if (wantBal < _amount) {
+            _deleverage();
+            wantBal = IERC20(want).balanceOf(address(this));
+        }
+
+        if (wantBal > _amount) {
+            wantBal = _amount;
+        }
+
+        if (tx.origin != owner() && !paused()) {
+            uint256 withdrawalFeeAmount = wantBal * withdrawalFee / WITHDRAWAL_MAX;
+            wantBal = wantBal - withdrawalFeeAmount;
+        }
+
+        IERC20(want).safeTransfer(vault, wantBal);
+        emit Withdraw(balanceOf());
+
+        if (!paused()) {
+            _leverage(availableWant());
+        }
+    }
+
+    /**
+     * @dev Required for various functions that need to deduct {reserves} from total {want}.
+     * @return how much {want} the contract holds without reserves
+     */
+    function availableWant() public view returns (uint256) {
+        uint256 wantBal = IERC20(want).balanceOf(address(this));
+        return wantBal - reserves;
+    }
+
+    // return supply and borrow balance
+    function updateBalance() public {
+        uint256 supplyBal = IVToken(iToken).balanceOfUnderlying(address(this));
+        uint256 borrowBal = IVToken(iToken).borrowBalanceCurrent(address(this));
+        balanceOfPool = supplyBal - borrowBal;
+    }
+
+
+    // calculate the total underlaying 'want' held by the strat.
+    function balanceOf() public view returns (uint256) {
+        return balanceOfWant() + balanceOfPool;
+    }
+
+    // it calculates how much 'want' this contract holds.
+    function balanceOfWant() public view returns (uint256) {
+        return IERC20(want).balanceOf(address(this));
+    }
+
+    // returns rewards unharvested
+    function rewardsAvailable() public returns (uint256) {
+        IComptroller(comptroller).claimComp(address(this), markets);
+        return IERC20(output).balanceOf(address(this));
+    }
+
+    // native reward amount for calling harvest
+    function callReward() public returns (uint256) {
+        IFeeConfig.FeeCategory memory fees = getFees();
+        uint256 outputBal = rewardsAvailable();
+        uint256 nativeOut;
+        if (outputBal > 0) {
+            (nativeOut,) = ISolidlyRouter(unirouter).getAmountOut(outputBal, output, native);
```

Comment:
```suggestion
            uint256[] memory amounts = ISolidlyRouter(unirouter).getAmountsOut(outputBal, outputToNativeRoute);
            return amounts[amounts.length-1];
```
callReward() was only checking direct route to native, not following the right route.  This incorrectly resulted in a return value of 0.